node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeadmin/NodeAdminStateUpdater.java

```diff
@@ -291,5 +314,11 @@ public void stop() {
 
         // Finally, stop NodeAdmin and all the NodeAgents
         nodeAdmin.stop();
+
+        log.info(objectToString() + ": Releasing lock");
+        if (classLock != null) {
```

Comment:
If the classLock is optional I suggest we make that clearer by using Optional

Reply:
Fixed.